src/systems/systems/src/air_conditioning/acs_controller.rs

```diff
@@ -0,0 +1,1808 @@
+
+impl AcState<EndTakeOff> {
+    fn step(
+        self: AcState<EndTakeOff>,
+        context: &UpdateContext,
+        lgciu: [&impl LgciuWeightOnWheels; 2],
+    ) -> AcStateManager {
+        if !lgciu
+            .iter()
+            .all(|&a| a.left_and_right_gear_compressed(true))
+            || self.timer > Duration::from_secs(10)
+        {
+            AcStateManager::InFlight(self.into())
+        } else {
+            AcStateManager::EndTakeOff(self.increase_timer(context))
+        }
+    }
+}
+
+transition!(EndTakeOff, InFlight);
+
+#[derive(Copy, Clone)]
+struct InFlight;
+
+impl AcState<InFlight> {
+    fn step(
+        self: AcState<InFlight>,
+        engines: [&impl EngineCorrectedN1; 2],
+        lgciu: [&impl LgciuWeightOnWheels; 2],
+    ) -> AcStateManager {
+        if engines
+            .iter()
+            .all(|&x| x.corrected_n1() < Ratio::new::<percent>(70.))
+            && lgciu
+                .iter()
+                .all(|&a| a.left_and_right_gear_compressed(true))
+        {
+            AcStateManager::BeginLanding(self.into())
+        } else {
+            AcStateManager::InFlight(self)
+        }
+    }
+}
+
+transition!(InFlight, BeginLanding);
+
+#[derive(Copy, Clone)]
+struct BeginLanding;
+
+impl AcState<BeginLanding> {
+    fn step(
+        self: AcState<BeginLanding>,
+        context: &UpdateContext,
+        engines: [&impl EngineCorrectedN1; 2],
+    ) -> AcStateManager {
+        if (engines
+            .iter()
+            .all(|&x| x.corrected_n1() < Ratio::new::<percent>(70.))
+            && context.indicated_airspeed().get::<knot>() < 70.)
+            || self.timer > Duration::from_secs(35)
+        {
+            AcStateManager::EndLanding(self.into())
+        } else {
+            AcStateManager::BeginLanding(self.increase_timer(context))
+        }
+    }
+}
+
+transition!(BeginLanding, EndLanding);
+
+#[derive(Copy, Clone)]
+struct EndLanding;
+
+impl AcState<EndLanding> {
+    fn step(self: AcState<EndLanding>, context: &UpdateContext) -> AcStateManager {
+        if self.timer > Duration::from_secs(10) {
+            AcStateManager::OnGround(self.into())
+        } else {
+            AcStateManager::EndLanding(self.increase_timer(context))
+        }
+    }
+}
+
+transition!(EndLanding, OnGround);
+
+#[derive(Clone)]
+struct ZoneController {
+    zone_temp_id: VariableIdentifier,
+    zone_duct_temp_id: VariableIdentifier,
+
+    zone_id: &'static str,
+    duct_demand_temperature: ThermodynamicTemperature,
+    zone_selected_temperature: ThermodynamicTemperature,
+    zone_measured_temperature: ThermodynamicTemperature,
+    pid_controller: PidController,
+}
+
+impl ZoneController {
+    const K_ALTITUDE_CORRECTION: f64 = 0.0000375; // deg/feet
+    const UPPER_DUCT_TEMP_LIMIT_LOW: f64 = 323.15; // K
+    const UPPER_DUCT_TEMP_LIMIT_HIGH: f64 = 343.15; // K
+    const LOWER_DUCT_TEMP_LIMIT_LOW: f64 = 275.15; // K
+    const LOWER_DUCT_TEMP_LIMIT_HIGH: f64 = 281.15; // K
```

Comment:
**suggestion**: I'd include the unit in the variable name.

Reply:
As per comment above, implemented.